aspnetcore/blazor/forms-and-input-components.md

```diff
@@ -37,28 +37,90 @@ The <xref:Microsoft.AspNetCore.Components.Forms?displayProperty=fullName> namesp
+
+To disable enhanced navigation for a form:
+
+* For an `EditForm`, remove the `Enhance` parameter from the form element (or set it to `false`: `Enhance="false"`).
+* For an HTML `<form>`, remove the `data-enhance` attribute from form element (or set it to `false`: `data-enhance="false"`).
+
+Blazor's enhanced navigation and form handing may undo dynamic changes to the DOM if the updated content isn't part of the server rendering. To preserve the content of an element, use the `data-permanent` attribute.
+
+In the following example, the content of the `<div>` element is updated dynamically by a script when the page loads:
+
+```html
+<div data-permanent>
+    ...
+</div>
+```
+
+To disable enhanced navigation and form handling globally, see <xref:blazor/fundamentals/startup#enhanced-navigation-and-form-handling>.
+
+Once Blazor has started on the client, you can use the `enhancedload` event to listen for enhanced page updates. This allows for re-applying changes to the DOM that may have been undone by an enhanced page update.
+
+```javascript
+Blazor.addEventListener('enhancedload', () => console.log('Enhanced update!'));
+```
+
+:::moniker-end
+
 ## Examples in this article
 
 :::moniker range=">= aspnetcore-8.0"
 
-Components are configured for interactivity with server rendering. For a client-side experience in a Blazor Web App, change the render mode in the `@attribute` directive at the top of the component to either:
+Components are configured for interactivity with server rendering and enhanced navigation. For a client-side experience in a Blazor Web App, change the render mode in the `@rendermode` directive at the top of the component to either:
+
+<!-- UPDATE 8.0 These simplify at RTM -->
 
-* `RenderModeWebAssembly` for interactive client rendering only.
-* `RenderModeAuto` for interactive client rendering after interactive server rendering, which operates while the Blazor app bundle downloads in the background and the .NET WebAssembly runtime starts on the client.
+* `RenderMode.InteractiveWebAssembly` for interactive client rendering only.
+* `RenderMode.InteractiveAuto` for interactive client rendering after interactive server rendering, which operates while the Blazor app bundle downloads in the background and the .NET WebAssembly runtime starts on the client.
 
-If working with a Blazor WebAssembly app, take ***either*** of the following approaches:
+If working with a standalone Blazor WebAssembly app, take ***either*** of the following approaches:
 
-* Change the render mode to `RenderModeWebAssembly`:
+* Change the render mode to `RenderMode.InteractiveWebAssembly`:
 
   ```diff
-  - @attribute [RenderModeServer]
-  + @attribute [RenderModeWebAssembly]
+  - @rendermode RenderMode.InteractiveServer
+  + @rendermode RenderMode.InteractiveWebAssembly
```

Comment:
Standalone Blazor WebAssembly apps don't use the render modes. The whole app is running interactively on WebAssembly, just like it always has. All of the interactive forms in this doc should work just fine in a standalone Blazor WebAssembly app as long as the code doesn't make any assumptions about running on the server instead of the client.